src/install.c

```diff
@@ -1225,11 +1222,14 @@ gboolean do_install_bundle(RaucInstallArgs *args, GError **error)
 	g_assert_null(r_context()->install_info->mounted_bundle);
 
 	r_context_begin_step("do_install_bundle", "Installing", 5);
+
+	r_context_begin_step("determine_slot_states", "Determining slot states", 0);
 	res = determine_slot_states(&ierror);
 	if (!res) {
 		g_propagate_error(error, ierror);
 		goto out;
 	}
+	r_context_end_step("determine_slot_states", res);
```

Comment:
Step will not be terminated in case of failure, also, `res` will always be TRUE here.